sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/controllers/ControllerExceptionHandler.java

```diff
@@ -0,0 +1,27 @@
+package com.sap.ai.sdk.app.controllers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import javax.annotation.Nonnull;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.ErrorResponse;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+/** Exception handler for the controllers. */
+@SuppressWarnings("unused")
+@RestControllerAdvice
+public class ControllerExceptionHandler {
```

Comment:
(Question)

What do you need this class for?

Reply:
As discussed, I will open a new PR with a class that handles more/all possible exceptions.